pymbolic/primitives.py

```diff
@@ -1669,15 +1669,13 @@ def make_common_subexpression(field, prefix=None, scope=None):
         # Don't re-wrap
         return field
 
-    try:
-        from pytools.obj_array import log_shape
-    except ImportError:
-        have_obj_array = False
+    if isinstance(field, numpy.ndarray):
+        if field.dtype.char == "O":
+            logical_shape = field.shape
+        else:
+            logical_shape = field.shape[:-1]
```

Comment:
This logic was taken from `pytools.obj_array.log_shape`, but I'm not actually sure why it strips the final dimension for non-object arrays.

Reply:
I changed this in 14718ade39aeb72e52604213ffb95806aac3f167